### DISCUSSION TITLE
Compress JPEGs

### DISCUSSION THREAD

**Discussion by mirisbowring** [sysadminsmedia/homebox#341] [Ideas]
Hi,
I started using homebox and added around 100 Items with an cover image.
The images have been taken with my iPhone 12 and uploaded via the mobile UI.

I experienced slow loading times in my basement (though that this is due to the sql lite DB).
It turned out (when i was inspecting this issue on the computer) that the images from the iPhone are around 5MB each. This means that the Website needs to load 20-100MB to load the overview. This causes slow loading times and high data usage.

I think it would be great to have thumbnails for the fotos.
Either by compressing the images on upload or by generating additional thumbnails that are just a few KBs in size since they are only small in the display anyways.

EDIT: Just found the issue for v1: https://github.com/sysadminsmedia/homebox/issues/272

**Comment by tankerkiller125** [comment]
I'm currently digging through the upload and storage code (part of the v1 Roadmap), image compression (or rather conversion) is on the list of things to do. Notably we intend to convert all images to webp (which is well supported, and much smaller than JPG and PNG). How we handle this though is something I'm still not certain on, given some people want the original to be stored as well, which may require some UI changes and other things.

**Reply by mirisbowring** [reply]
I think, this one is easy.
You would keep both per default.
In the Directory it would look like this:

```bash
abc-def-ghi.jpg
abc-def-ghi_thumb.webp
```

In the UI you would always load the `..._thumb.webp` file, EXCEPT if the user clicks on the attachment to enlarge the image. The you would load the original and the user would also be able to download the original.

**Reply by mirisbowring** [reply]
Maybe some users would like to have an env var or something to disable the originals and keep the thumbnails only - idk.

**Comment by 404invalid-user** [comment]
ha I was also just thinking about this I have taken a photo of all my devices so load times are very slow

**Comment by tankerkiller125** [comment]
Closing as part of consolidation into #265 which will have image manipulation like compression and format changes and previews.
